src/shared/allowed-patterns.js

```diff
@@ -1,5 +1,6 @@
 // shared/allowed-patterns.js
 const ALLOWED_PATTERNS = [
+  /https:\/\/www\.amazon\..*?\/(?:dp|gp\/product)\/.*?(B[\dA-Z]{9}|\d{9}(?:X|\d))/,
   /https:\/\/www\.amazon\.[a-z.]+\/(?:gp\/product|dp|[^/]+\/dp)\/[A-Z0-9]{10}/,
   /https:\/\/www\.amazon\.[a-z.]+\/[^/]+\/dp\/[A-Z0-9]{10}/,
   /https:\/\/www\.amazon\.[a-z.]+\/-\/[a-z]+\/[^/]+\/dp\/[A-Z0-9]{10}/, // for paths with language segments
```

Comment:
These 3 can probably be safely removed, but they don't affect anything so they can be left in